modules/masterbar/masterbar.php

```diff
@@ -0,0 +1,821 @@
+				'url'   => 'https://wordpress.com/types/jetpack-portfolio/' . esc_attr( $this->primary_site_slug ),
+				'id'    => 'wp-admin-bar-edit-portfolio',
+				'label' => __( 'Portfolio', 'jetpack' ),
+			),
+			array(
+				'url'   => 'https://wordpress.com/edit/jetpack-portfolio/' . esc_attr( $this->primary_site_slug ),
+				'id'    => 'wp-admin-bar-new-portfolio',
+				'label' => _x( 'Add', 'admin bar menu new item label', 'jetpack' ),
+			)
+		);
+
+		$wp_admin_bar->add_menu( array(
+			'parent' => 'publish',
+			'id'     => 'new-jetpack-portfolio',
+			'title'  => $portfolios_title,
+			'meta'   => array(
+				'class' => 'inline-action',
+			),
+		) );
+
+		if ( current_user_can( 'edit_theme_options' ) ) {
+			// Look and Feel group
+			$wp_admin_bar->add_group( array(
+				'parent' => 'blog',
+				'id'     => 'look-and-feel',
+			) );
+
+			// Look and Feel header
+			$wp_admin_bar->add_menu( array(
+				'parent' => 'look-and-feel',
+				'id'     => 'look-and-feel-header',
+				'title'  => _x( 'Personalize', 'admin bar menu group label', 'jetpack' ),
+				'meta'   => array(
+					'class' => 'ab-submenu-header',
+				),
+			) );
+
+			if ( is_admin() ) {
+				// In wp-admin the `return` query arg will return to that page after closing the Customizer
+				$customizer_url = add_query_arg( array( 'return' => urlencode( site_url( $_SERVER['REQUEST_URI'] ) ) ), wp_customize_url() );
+			} else {
+				// On the frontend the `url` query arg will load that page in the Customizer and also return to it after closing
+				// non-home URLs won't work unless we undo domain mapping since the Customizer preview is unmapped to always have HTTPS
+				global $current_blog;
+				$current_page = 'https://' . $current_blog->domain . $_SERVER['REQUEST_URI'];
+				$customizer_url = add_query_arg( array( 'url' => urlencode( $current_page ) ), wp_customize_url() );
+			}
+
+			$theme_title = $this->create_menu_item_pair(
+				array(
+					'url'   => 'https://wordpress.com/design/' . esc_attr( $this->primary_site_slug ),
+					'id'    => 'wp-admin-bar-themes',
+					'label' => __( 'Themes', 'jetpack' ),
+				),
+				array(
+					'url'   => $customizer_url,
+					'id'    => 'wp-admin-bar-cmz',
+					'label' => _x( 'Customize', 'admin bar customize item label', 'jetpack' ),
+				)
+			);
+			$meta = array( 'class' => 'mb-icon', 'class' => 'inline-action' );
+			$href = false;
+
+			$wp_admin_bar->add_menu( array(
+				'parent' => 'look-and-feel',
+				'id'     => 'themes',
+				'title'  => $theme_title,
+				'href'   => $href,
+				'meta'   => $meta
+			) );
+
+			if ( current_theme_supports( 'menus' ) ) {
+				$wp_admin_bar->add_menu( array(
+					'parent' => 'look-and-feel',
+					'id'     => 'menus',
+					'title'  => __( 'Menus', 'jetpack' ),
+					'href'   => 'https://wordpress.com/menus/' . esc_attr( $this->primary_site_slug ),
+					'meta' => array(
+						'class' => 'mb-icon',
+					),
+				) );
+			}
+		}
+
+		if ( current_user_can( 'manage_options' ) ) {
```

Comment:
In case their account is not connected we will bail in constructor, and the masterbar won't be shown at all (https://github.com/Automattic/jetpack/pull/6413/files#diff-d73a71fc4f0617de9989f4321bc4f39cR20).